pkg/adapter/configError.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package adapter
+
+import (
+	"fmt"
+	me "github.com/hashicorp/go-multierror"
+)
+
+// ConfigError represents an error encountered while validating a block of configuration state.
+type ConfigError struct {
+	Field      string
+	Underlying error
+}
+
+// Append adds a ConfigError to a multierror. This function is intended
```

Comment:
why not expose a NewConfigError() method (or even ConfigErrorf()) that took field, format, and args and then use the existing Append() method?

`func NewConfigError(field, format string, args... interface{}) ConfigError {...}`

or

`func ConfigErrorf(field, format string, args... interface{})`

The calling code could then be something like:

`me.Append(adapter.ConfigErrorf("field1", "Bad ...: %s", fieldVal))`

Which strikes me as more readable (and closer to how fmt.Errorf(), etc., work).

Also, configError.go wouldn't have to import go-multierror.